### ISSUE TITLE
Consider dyncall implementation

### ISSUE THREAD

**Issue by plicease** [plicease/FFI-Platypus#76]
http://www.dyncall.org/

may be preferred in some environments over libffi (?).  it's a question.

**Comment by plicease** [comment]
I discovered dyncall from looking at the Perl 6 source.  You can correct me if I am wrong, but I think both dyncall and libffi are supported, the original was dyncall and libffi was added for platforms that didn't support dyncall.

From my perspective having multiple implementations would insulate `FFI::Platypus` from being dependent on a single implementation.  libffi is pretty good but they haven't had a release in years.  The most recent release doesn't build correctly on windows (`Alien::FFI` has some logic to use the previous release on windows).  They are threatening to do another release :crossed_fingers: though :)

**Comment by mohawk2** [comment]
It looks like libffi's latest release is 3.4.2 from Jun 2021, if that makes a difference.

**Comment by plicease** [comment]
> It looks like libffi's latest release is 3.4.2 from Jun 2021, if that makes a difference.

Yes some of the commentary here is out of date, but I still think there is some potentially utility on having an alternate implementation.
